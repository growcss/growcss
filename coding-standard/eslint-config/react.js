module.exports ={
  extends: [
    'prettier/react',
    'plugin:jest/recommended'
  ],
  plugins: [
    'react',
    'jsx-a11y',
    // 'security', @todo check it on the next version
  ],
  rules: {
    ...{
      'react/sort-comp': 'off',
      'react/jsx-filename-extension': 'off',
      'react/require-default-props': 'off',
    },
    ...require('./rules/prettier-conflicts-react'),
    ...require('./rules/jest'),
  },
  env: {
    browser: true,
    commonjs: true,
    es6: true,
    jest: true,
    node: true,
  },
};
