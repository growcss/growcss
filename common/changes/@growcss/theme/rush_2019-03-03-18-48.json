{
  "changes": [
    {
      "packageName": "@growcss/theme",
      "comment": "feat(theme): added new theme component",
      "type": "major"
    }
  ],
  "packageName": "@growcss/theme",
  "email": "d.bannert@anolilab.de"
}
