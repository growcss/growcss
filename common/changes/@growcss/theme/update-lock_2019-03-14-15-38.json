{
  "changes": [
    {
      "comment": "",
      "packageName": "@growcss/theme",
      "type": "none"
    }
  ],
  "packageName": "@growcss/theme",
  "email": "d.bannert@anolilab.de"
}