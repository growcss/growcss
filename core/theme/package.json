{
  "name": "@growcss/theme",
  "version": "0.0.0",
  "description": "The GrowCss theme framework.",
  "keywords": [
    "css",
    "growcss",
    "react",
    "react-component",
    "reactjs",
    "styled-components",
    "theme",
    "ui"
  ],
  "homepage": "http://growcss.com/packages/theme",
  "repository": "https://github.com/growcss/growcss.git",
  "license": "MIT",
  "author": {
    "name": "Daniel Bannert",
    "email": "d.bannert@anolilab.de"
  },
  "files": [
    "dist",
    "README.md",
    "CHANGELOG.md",
    "LICENSE"
  ],
  "main": "dist/index.cjs.js",
  "unpkg": "dist/index.umd.js",
  "module": "dist/index.es.js",
  "browser": "dist/umd/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "                     npm run clear && ./node_modules/.bin/growcss-rollup",
    "clear": "                     ./node_modules/@build/scripts/node_modules/.bin/rimraf dist coverage",
    "install": "                   node bin/create-normalize-file.js",
    "lint": "                      npm run lint:eslint:source && npm run lint:eslint:supporting && npm run lint:style",
    "lint-fix": "                  npm run lint:fix:eslint:source && npm run lint:fix:eslint:supporting",
    "lint:eslint:source": "        eslint -c ./node_modules/@build/extended-eslint-config/.eslintrc.sources.js --print-config ./node_modules/@build/extended-eslint-config/.eslintrc.sources.js | eslint-config-prettier-check && eslint -c ./node_modules/@build/extended-eslint-config/.eslintrc.sources.js './src/**/*.{js,jsx,ts,tsx}'",
    "lint:eslint:supporting": "    eslint -c ./node_modules/@build/extended-eslint-config/.eslintrc.supporting.js --print-config ./node_modules/@build/extended-eslint-config/.eslintrc.supporting.js | eslint-config-prettier-check && eslint -c ./node_modules/@build/extended-eslint-config/.eslintrc.supporting.js '{__tests__,__stories__}/**/*.{js,jsx,ts,tsx}'",
    "lint:fix:eslint:source": "    npm run lint:eslint:source -- --fix",
    "lint:fix:eslint:supporting": "npm run lint:eslint:supporting -- --fix",
    "lint:style": "                stylelint-config-prettier-check && stylelint './src/**/*.{js,ts,tsx}'",
    "sort": "                      ./node_modules/@build/scripts/node_modules/.bin/sort-package-json",
    "test": "                      cross-env NODE_ENV=test jest --config=./node_modules/@build/jest/jest.config.js --json --outputFile=.jest-test-results.json || true",
    "test-ci": "                   cross-env NODE_ENV=test jest --config=./node_modules/@build/jest/jest.config.js --runInBand --ci",
    "test-ci-coverage": "          cross-env NODE_ENV=test jest --config=./node_modules/@build/jest/jest.config.js --coverage --runInBand --ci",
    "test-coverage": "             cross-env NODE_ENV=test jest --config=./node_modules/@build/jest/jest.config.js --coverage",
    "test-snapshot-update": "      cross-env NODE_ENV=test jest --config=./node_modules/@build/jest/jest.config.js -u"
  },
  "dependencies": {
    "@growcss/elaborate": "^0.0.0"
  },
  "devDependencies": {
    "@build/extended-babel": "0.0.0",
    "@build/extended-eslint-config": "0.0.0",
    "@build/jest": "0.0.0",
    "@build/rollup": "0.0.0",
    "@build/scripts": "0.0.0",
    "@growcss/stylelint-config": "^1.0.2",
    "@types/node": "^11.9.4",
    "@types/react": "^16.8.3",
    "@types/react-dom": "^16.8.2",
    "@types/storybook__react": "^4.0.1",
    "@types/styled-components": "^4.1.10",
    "normalize.css": "^8.0.1",
    "react": "^16.8.2",
    "styled-components": "^4.1.3",
    "stylis": "^3.5.4"
  },
  "peerDependencies": {
    "react": "^16.8.0",
    "styled-components": "^4.0.1"
  },
  "engines": {
    "node": "^10.14.0 || ^11.3.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "moduleName": "theme"
}
