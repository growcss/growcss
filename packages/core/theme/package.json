{
  "name": "@growcss/theme",
  "version": "1.0.0",
  "description": "The GrowCss theme framework.",
  "keywords": [
    "css",
    "growcss",
    "react",
    "react-component",
    "reactjs",
    "styled-components",
    "theme",
    "ui"
  ],
  "homepage": "http://growcss.com/packages/theme",
  "repository": "https://github.com/growcss/growcss.git",
  "license": "MIT",
  "author": {
    "name": "Daniel Bannert",
    "email": "d.bannert@anolilab.de"
  },
  "files": [
    "dist",
    "README.md",
    "CHANGELOG.md",
    "LICENSE"
  ],
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "jsnext:main": "dist/es/index.js",
  "browser": "dist/umd/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "cross-env PACKAGE_PATH=core/theme rollup -c ../../../build/rollup.config.js",
    "install": "node bin/create-normalize-file.js"
  },
  "dependencies": {
    "@growcss/elaborate": "0.0.3"
  },
  "devDependencies": {
    "normalize.css": "^8.0.1"
  },
  "peerDependencies": {
    "react": "^16.0",
    "styled-components": "^4.0.1"
  },
  "engines": {
    "node": "^10.14.0 || ^11.3.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "moduleName": "theme",
  "unpkg": "dist/umd/index.js"
}
