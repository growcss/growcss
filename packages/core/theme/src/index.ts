import { createGlobalStyle } from 'styled-components';
import { h1, h2, h3, h4, h5, h6, content } from './typography';
import { colors } from './colors';
import normalize from './modern-normalize';

/**
 * These theme values are expressed as functions so that if we decide to make
 * them dependent on props in the future, it wouldn't require a significant
 * refactor everywhere they are being used.
 */
export const GrowCss = {
  theme: {
    colors,
    typography: {
      baseFontSize: () => '16px',
      fontFamily: () =>
        '-apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen", "Ubuntu", "Fira Sans", "Droid Sans", "Helvetica Neue", sans-serif',
      codeFontFamily: () =>
        '"SFMono-Medium", "SF Mono", "Segoe UI Mono", "Roboto Mono", "Ubuntu Mono", Menlo, Consolas, Courier, monospace',
      h1: () => h1,
      h2: () => h2,
      h3: () => h3,
      h4: () => h4,
      h5: () => h5,
      h6: () => h6,
    },
    grid: {
      gutterSize: (multiplier: number | undefined = undefined) => 10 * (multiplier || 1),
    },
    screens: {},
    shadows: {
      default: '0 2px 4px 0 rgba(0,0,0,0.10)',
      md: '0 4px 8px 0 rgba(0,0,0,0.12), 0 2px 4px 0 rgba(0,0,0,0.08)',
      lg: '0 15px 30px 0 rgba(0,0,0,0.11), 0 5px 15px 0 rgba(0,0,0,0.08)',
      inner: 'inset 0 2px 4px 0 rgba(0,0,0,0.06)',
      outline: '0 0 0 3px rgba(52,144,220,0.5)',
      none: 'none',
    },
    zIndex: {
      auto: 'auto',
      '0': 0,
      '10': 10,
      '20': 20,
      '30': 30,
      '40': 40,
      '50': 50,
    },
    spinner: {},
    image: {},
  },
};

export const GlobalStyle = createGlobalStyle`
  ${normalize}
  
  body,
  html {
    height: 100%;
    width: 100%;
  }
  
  body {
    background-color: ${GrowCss.theme.colors.white};
    font-family: ${GrowCss.theme.typography.fontFamily};
    ${content}
    margin-bottom: 0; // overwrite the font margin
    font-style: normal;
    font-weight: 400;
    -ms-overflow-style: -ms-autohiding-scrollbar;
    text-decoration-skip: ink;
  }
  
  a {
    color: ${GrowCss.theme.colors.blue};
    text-decoration: none;
  }
  a:hover {
    color: ${GrowCss.theme.colors.blueLight};
    text-decoration: underline;
  }
  a:active {
    color: ${GrowCss.theme.colors.blueDark};
  }
  a:focus {
    outline: 2px solid ${GrowCss.theme.colors.blueLighter};
    outline-offset: 2px;
  }

  p,
  ul,
  ol,
  dl,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  blockquote,
  pre,
  form,
  table {
    margin: ${GrowCss.theme.grid.gutterSize(1.5)}px 0 0 0;
  }
  
  h1 {
    ${GrowCss.theme.typography.h1}
  }
  h2 {
    ${GrowCss.theme.typography.h2}
  }
  h3 {
    ${GrowCss.theme.typography.h3}
  }
  h4 {
    ${GrowCss.theme.typography.h4}
  }
  h5 {
    ${GrowCss.theme.typography.h5}
  }
  h6 {
    ${GrowCss.theme.typography.h6}
  }
  
  code,
  kbd {
    font-family: ${GrowCss.theme.typography.codeFontFamily};
  }

  var,
  address,
  dfn,
  cite {
    font-style: italic;
  }

  abbr {
    border-bottom: 1px ${GrowCss.theme.colors.grey} dotted;
    cursor: help;
  }
  
  blockquote,
  q {
    color: inherit;
  }
  blockquote {
    border: none;
    padding-left: ${GrowCss.theme.grid.gutterSize(5)}px;
  }
  [dir='rtl'] blockquote {
    padding-left: 0;
    padding-right: ${GrowCss.theme.grid.gutterSize(5)}px;
  }
  blockquote::before,
  q::before {
    content: '\\\\201C';
  }
  blockquote::after,
  q::after {
    content: '\\\\201D';
  }
  blockquote::before {
    float: left;
    /* to keep the quotes left of any child elements like blockquote > p */
    margin-left: -1em;
    text-align: right;
    width: 1em;
  }
  [dir='rtl'] blockquote::before {
    float: right;
    margin-right: -1em;
    text-align: left;
  }
  blockquote > :last-child {
    display: inline-block; /* so the quotes added via pseudos follow it immediately. */
  }
  
  p,
  ul,
  ol,
  dl,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  blockquote,
  pre,
  form,
  table {
    margin-top: ${GrowCss.theme.grid.gutterSize(1.5)}px;
  }
`;
