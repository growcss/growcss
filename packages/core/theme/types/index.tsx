import { DefaultTheme, FlattenSimpleInterpolation } from 'styled-components';

export interface GrowCssTheme extends DefaultTheme {
  global: {
    margin: string;
    padding: string;
    position: string;
    textDirection: string;
    printBreakpoint: string;
    stdWebDpi: number;
    breakpoints: {
      [key: string]: number;
    };
    hidpiBreakpoints: {
      [key: string]: number;
    };
  };
  body: {
    backgroundColor: string;
  };
  colors: {
    [key: string]: string;
  };
  typography: {
    color: string;
    fontSize: string;
    lineHeight: number;
    weightNormal: string | number;
    weightBold: string;
    antialiased: boolean;
    fontFamily: string;
    codeFontFamily: string;
    breakpoints: {
      [key: string]: number[];
    };
    h1?: FlattenSimpleInterpolation;
    h2?: FlattenSimpleInterpolation;
    h3?: FlattenSimpleInterpolation;
    h4?: FlattenSimpleInterpolation;
    h5?: FlattenSimpleInterpolation;
    h6?: FlattenSimpleInterpolation;
  };
  grid: {
    maxWidth: string;
    marginGutters: {
      [key: string]: string;
    };
    paddingGutters: {
      [key: string]: string;
    };
    columns: number;
  };
  shadows: any;
  zIndex: any;
  spinner: any;
  image: {
    breakpoints: {
      [key: string]: string;
    };
  };
}
