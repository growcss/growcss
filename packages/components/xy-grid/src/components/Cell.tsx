import * as React from 'react';
import classNames from 'classnames';
import { Gutters as DefaultGutters, GuttersProps } from './Gutters';
import { CellElement } from '../styled/CellElement';

export interface CellProps extends React.HTMLAttributes<HTMLDivElement>{
  children: React.ReactNode,
  gridColumns?: number,
  vertical?: boolean,
  gutterSizes?: GuttersProps,
  cellType?: string,
  gutterType?: string,
  align?: string,
}

export interface OffsetProps {
  smallOffset?: number,
  mediumOffset?: number,
  largeOffset?: number,
  xlargeOffset?: number,
  xxlargeOffset?: number,
}

export interface CellBreakpointsProps {
  small?: number | string,
  medium?: number | string,
  large?: number | string,
  xlarge?: number | string,
  xxlarge?: number | string,
}

export default class Cell extends React.Component<CellProps & CellBreakpointsProps & OffsetProps> {
  static defaultProps = {
    gridColumns: 12,
    gutterSizes: DefaultGutters,
    vertical: false,
  };

  render() {
    const {
      children,
      gridColumns,
      cellType,
      gutterType,
      gutterSizes,
      vertical,
      align,
      ...other
    } = this.props;
    const className = classNames('gc-cell');

    return (
      <CellElement
        className={className}
        gridColumns={gridColumns!}
        cellType={cellType}
        gutterType={gutterType}
        gutterSizes={gutterSizes!}
        vertical={vertical!}
        align={align}
        {...other}
      >
        {children}
      </CellElement>
    );
  }
}
