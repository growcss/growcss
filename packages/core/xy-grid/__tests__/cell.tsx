import * as React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { GrowCss } from '@growcss/theme';
import Cell from '../src/components/cell';

test('cell base layout', () => {
  const container = <Cell theme={GrowCss}>test</Cell>;
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex', '0 0 auto');
  expect(tree).toHaveStyleRule('width', '100%');
  expect(tree).toHaveStyleRule('min-height', '0px');
  expect(tree).toHaveStyleRule('min-width', '0px');
});

test('cell auto layout with padding gutter', () => {
  const container = (
    <Cell theme={GrowCss} cellType="auto" gutterType="padding">
      test
    </Cell>
  );
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex', '1 1 0px');
  expect(tree).toHaveStyleRule('width', 'auto');
  expect(tree).toHaveStyleRule('min-height', '0px');
  expect(tree).toHaveStyleRule('min-width', '0px');
  expect(tree).toHaveStyleRule('box-sizing', 'border-box');
});

test('cell auto layout with margin gutter', () => {
  const container = (
    <Cell theme={GrowCss} cellType="auto" gutterType="margin">
      test
    </Cell>
  );
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex', '1 1 0px');
  expect(tree).toHaveStyleRule('width', 'auto');
  expect(tree).toHaveStyleRule('min-height', '0px');
  expect(tree).toHaveStyleRule('min-width', '0px');
});

test('cell shrink layout', () => {
  const container = (
    <Cell theme={GrowCss} cellType="shrink">
      test
    </Cell>
  );
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex', '0 0 auto');
  expect(tree).toHaveStyleRule('width', 'auto');
  expect(tree).toHaveStyleRule('min-height', '0px');
  expect(tree).toHaveStyleRule('min-width', '0px');
});

test('cell with 6 small layout', () => {
  const container = (
    <Cell theme={GrowCss} small={6}>
      test
    </Cell>
  );
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex', '0 0 auto');
  expect(tree).toHaveStyleRule('min-height', '0px');
  expect(tree).toHaveStyleRule('min-width', '0px');
  expect(tree).toHaveStyleRule('width', '50%');
  expect(tree).toHaveStyleRule('flex-basis', 'auto');
});

test('cell align', () => {
  const container = (
    <Cell theme={GrowCss} align="top">
      test
    </Cell>
  );
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex', '0 0 auto');
  expect(tree).toHaveStyleRule('width', '100%');
  expect(tree).toHaveStyleRule('min-height', '0px');
  expect(tree).toHaveStyleRule('min-width', '0px');
  expect(tree).toHaveStyleRule('align-self', 'flex-start');
});

test('cell offset', () => {
  const container = (
    <Cell theme={GrowCss} smallOffset={1}>
      test
    </Cell>
  );
  const tree = renderer.create(container).toJSON();

  expect(shallow(container)).toMatchSnapshot();
  expect(tree).toHaveStyleRule('flex', '0 0 auto');
  expect(tree).toHaveStyleRule('width', '100%');
  expect(tree).toHaveStyleRule('min-height', '0px');
  expect(tree).toHaveStyleRule('min-width', '0px');
  expect(tree).toHaveStyleRule('margin-left', '8.333333333333334%');
});
