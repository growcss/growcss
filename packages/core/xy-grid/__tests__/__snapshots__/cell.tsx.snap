// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cell align 1`] = `
<cell-element__CellElement
  align="top"
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  vertical={false}
>
  test
</cell-element__CellElement>
`;

exports[`cell auto layout with margin gutter 1`] = `
<cell-element__CellElement
  cellType="auto"
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  gutterType="margin"
  vertical={false}
>
  test
</cell-element__CellElement>
`;

exports[`cell auto layout with padding gutter 1`] = `
<cell-element__CellElement
  cellType="auto"
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  gutterType="padding"
  vertical={false}
>
  test
</cell-element__CellElement>
`;

exports[`cell base layout 1`] = `
<cell-element__CellElement
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  vertical={false}
>
  test
</cell-element__CellElement>
`;

exports[`cell offset 1`] = `
<cell-element__CellElement
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  smallOffset={1}
  vertical={false}
>
  test
</cell-element__CellElement>
`;

exports[`cell shrink layout 1`] = `
<cell-element__CellElement
  cellType="shrink"
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  vertical={false}
>
  test
</cell-element__CellElement>
`;

exports[`cell with 6 small layout 1`] = `
<cell-element__CellElement
  className="gc-cell"
  gridColumns={12}
  gutterSizes={
    Object {
      "medium": "30px",
      "small": "20px",
    }
  }
  small={6}
  vertical={false}
>
  test
</cell-element__CellElement>
`;
